tests/test_export_onnx.py

```diff
@@ -0,0 +1,114 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+
+import io
+import os
+import unittest
+import torch
+
+from torch.hub import _check_module_exists
+
+from detectron2 import model_zoo
+from detectron2.config import get_cfg
+from detectron2.export import add_export_config
+from detectron2.export.flatten import TracingAdapter
+from detectron2.modeling import build_model
+from detectron2.utils.env import TORCH_VERSION
+from detectron2.utils.testing import (
+    get_sample_coco_image,
+    SLOW_PUBLIC_CPU_TEST)
+
+
+SUPPORTED_ONNX_OPSET = 14
+
+
+@unittest.skipIf(not _check_module_exists("onnx"), "ONNX not installed.")
+class TestONNXTracingExport(unittest.TestCase):
+    def testMaskRCNNFPN(self):
+        def inference_func(model, images):
+            inputs = [{"image": image} for image in images]
+            inst = model.inference(inputs, do_postprocess=False)[0]
+            return [{"instances": inst}]
+
+        self._test_model_zoo_from_config_path(
+            "COCO-InstanceSegmentation/mask_rcnn_R_50_FPN_3x.yaml", inference_func
+        )
+
+    @SLOW_PUBLIC_CPU_TEST
+    def testMaskRCNNC4(self):
+        def inference_func(model, image):
+            inputs = [{"image": image}]
+            return model.inference(inputs, do_postprocess=False)[0]
+
+        self._test_model_zoo_from_config_path(
+            "COCO-InstanceSegmentation/mask_rcnn_R_50_C4_3x.yaml", inference_func
+        )
+
+    @SLOW_PUBLIC_CPU_TEST
+    def testCascadeRCNN(self):
+        def inference_func(model, image):
+            inputs = [{"image": image}]
+            return model.inference(inputs, do_postprocess=False)[0]
+
+        self._test_model_zoo_from_config_path(
+            "Misc/cascade_mask_rcnn_R_50_FPN_3x.yaml", inference_func
+        )
+
+    # bug fixed by https://github.com/pytorch/pytorch/pull/67734
+    @unittest.skipIf(TORCH_VERSION == (1, 10) and os.environ.get("CI"), "1.10 has bugs.")
+    def testRetinaNet(self):
+        def inference_func(model, image):
+            return model.forward([{"image": image}])[0]["instances"]
+
+        self._test_model_zoo_from_config_path(
+            "COCO-Detection/retinanet_R_50_FPN_3x.yaml", inference_func
+        )
+
+    def _test_model(self, model, inputs):
+        import onnx  # noqa: F401
+        f = io.BytesIO()
+        with torch.no_grad():
+            torch.onnx.export(
+                model,
+                inputs,
+                f,
+                training=torch.onnx.TrainingMode.EVAL,
+                operator_export_type=torch.onnx.OperatorExportTypes.ONNX,
+                opset_version=SUPPORTED_ONNX_OPSET,
```

Comment:
Could you also make necessary changes to https://github.com/facebookresearch/detectron2/blob/0ad20f13e23f2f4454be6196c1cd38e2171b294c/tools/deploy/export_model.py#L127 such that the example will work? It's likely not in a working state

Reply:
It does work, as long as `onnx` and `onnxoptimizer` are installed